### PR TITLE
Implement defaults down at FFI layer

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Previously many of the rust wrapper defaults were implemented in the
wrapper modules.  Now the FFI types implement the default
implementations and the rust wrappers mainly derive `Default`.

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #91** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/91" target="_blank">Graphite</a> 
    * **PR #86** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/86" target="_blank">Graphite</a> 
      * **PR #92** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/92" target="_blank">Graphite</a> 
        * **PR #103** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/103" target="_blank">Graphite</a> 
          * **PR #110** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/110" target="_blank">Graphite</a> 
            * **PR #111** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/111" target="_blank">Graphite</a> 
              * **PR #112** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/112" target="_blank">Graphite</a> 
                * **PR #113** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/113" target="_blank">Graphite</a> 
                  * **PR #115** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/115" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/115?utm_source=stack-comment).

File: core/types/src/svn.rs [hunk 1]
```diff
@@ -59,6 +59,7 @@ impl ConfigId {
     }
 }
 
+// Pure array type larger than 32 so must implement default at the newtype level
```
Comment (nick-mobilecoin) [inline core/types/src/svn.rs:62]: Since this is currently the only pure array wrapped, I chose not to do a macro yet...

**Review by awygle — APPROVED** [review]
I don't have an issue with the implementation here, but I also don't really understand the motivation for this PR. My belief is that once we have idiomatic Rust wrappers around all the FFI types, users should never interact with the bindgen-generated FFI types, meaning a Default implementation is of questionable utility at this level. Regardless, it certainly doesn't hurt to have it, and these changes look fine to me.

**Comment by nick-mobilecoin** [comment]
> I don't have an issue with the implementation here, but I also don't really understand the motivation for this PR. My belief is that once we have idiomatic Rust wrappers around all the FFI types, users should never interact with the bindgen-generated FFI types, meaning a Default implementation is of questionable utility at this level. Regardless, it certainly doesn't hurt to have it, and these changes look fine to me.

Two advantages I can think of:
- Having the defaults down here lets the rust types derive default, and keeps the boiler plate code out of them to focus on the main functionality
- For C interfaces that provide the outputs by pointer, one can do `let mut my_instance = some_c_type::default()` instead of manually initializing or using [MaybeUninit](https://doc.rust-lang.org/stable/core/mem/union.MaybeUninit.html)

**Review by jcape — APPROVED** [review]

File: core/sys/types/src/lib.rs [hunk 2]
```diff
@@ -10,3 +10,94 @@
...
+
+impl Default for sgx_ql_att_key_id_t {
+    fn default() -> Self {
+        Self {
+            id: Default::default(),
+            version: Default::default(),
+            // This is supposed to represent the length of the hash as SHA256
+            // or SHA384.  SHA384 is currently not supported by Intel so default
+            // to 32 or SHA256
+            mrsigner_length: 32,
```
[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply core/sys/types/src/lib.rs]: changed default to 0

[the remark this answers is not in the document]
Comment (nick-mobilecoin) [reply core/sys/types/src/lib.rs]: changed default to 0

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/115?utm_source=stack-comment).


Job Reference: C6ZyZ0Ggec8V0ksW3mYN

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: ⏳ Queued to merge

This pull request is currently queued to be merged as part of a stack.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/115?utm_source=stack-comment).


Job Reference: ANN5tbEFcYrOel5JRpoC

**Pull request merged**